mapflow/functional/layer_utils.py

```diff
@@ -504,7 +504,7 @@ def download_results_callback(self, response: QNetworkReply, processing) -> None
         self.dlg.processingsTable.setEnabled(True)
         # Avoid overwriting existing files by adding (n) to their names
         output_path = os.path.join(self.dlg.outputDirectory.text(), processing.id_)
-        extension = '.gpkg'
+        extension = '.geojson'
```

Comment:
We NEED to use gpkg whenever possible.
- it has internal compression and takes less memory
- it has internal indexing, so QGIS does handle large files, whereas with geojson it hangs and freezes.

So, we should fix the problem inside the data which prevents us from using gpkg, maybe it is in the backend / vector-exporter?

Anyway, if we can't fix the data here before saving, we can save it in geojson, it's a good fallback strategy. But we need to try saving as gpkg first, and use it only in case of fail.